frontend/packages/ceph-storage-plugin/src/utils/block-pool.ts

```diff
@@ -0,0 +1,142 @@
+import { apiVersionForModel } from '@console/internal/module/k8s';
+
+import { StoragePoolKind } from '../types';
+import { CephBlockPoolModel } from '../models';
+import { CEPH_STORAGE_NAMESPACE } from '../constants/index';
+import { COMPRESSION_ON, ROOK_MODEL } from '../constants/storage-pool-const';
+
+export type BlockPoolState = {
+  poolName: string;
+  poolStatus: string;
+  replicaSize: string;
+  isCompressed: boolean;
+  isArbiterCluster: boolean;
+  volumeType: string;
+  isSubmitted: boolean;
+  inProgress: boolean;
+  errorMessage: string;
+};
+
+export enum BlockPoolActionType {
+  SET_POOL_NAME = 'SET_POOL_NAME',
+  SET_POOL_STATUS = 'SET_POOL_STATUS',
+  SET_POOL_REPLICA_SIZE = 'SET_POOL_REPLICA_SIZE',
+  SET_POOL_COMPRESSED = 'SET_POOL_COMPRESSED',
+  SET_POOL_ARBITER = 'SET_POOL_ARBITER',
+  SET_POOL_VOLUME_TYPE = 'SET_POOL_VOLUME_TYPE',
+  SET_IS_SUBMITTED = 'SET_IS_SUBMITTED',
+  SET_INPROGRESS = 'SET_INPROGRESS',
+  SET_ERROR_MESSAGE = 'SET_ERROR_MESSAGE',
+}
+
+export type BlockPoolAction =
+  | { type: BlockPoolActionType.SET_POOL_NAME; payload: string }
+  | { type: BlockPoolActionType.SET_POOL_STATUS; payload: string }
+  | { type: BlockPoolActionType.SET_POOL_REPLICA_SIZE; payload: string }
+  | { type: BlockPoolActionType.SET_POOL_COMPRESSED; payload: boolean }
+  | { type: BlockPoolActionType.SET_POOL_ARBITER; payload: boolean }
+  | { type: BlockPoolActionType.SET_POOL_VOLUME_TYPE; payload: string }
+  | { type: BlockPoolActionType.SET_IS_SUBMITTED; payload: boolean }
+  | { type: BlockPoolActionType.SET_INPROGRESS; payload: boolean }
+  | { type: BlockPoolActionType.SET_ERROR_MESSAGE; payload: string };
+
+export const blockPoolInitialState: BlockPoolState = {
+  poolName: 'sc-pool',
```

Comment:
We should provide placeholder rather than the name i.e sc-pool and value of placeholder should be Name or pool name
```suggestion
  poolName: '',
```

Reply:
ack